indexing-service/src/main/java/io/druid/indexing/common/task/HadoopConverterTask.java

```diff
@@ -0,0 +1,276 @@
+    );
+    this.hadoopDependencyCoordinates = hadoopDependencyCoordinates;
+    this.distributedSuccessCache = Preconditions.checkNotNull(distributedSuccessCache, "distributedSuccessCache");
+    this.segmentOutputPath = Preconditions.checkNotNull(segmentOutputPath, "segmentOutputPath");
+    this.jobPriority = jobPriority;
+    this.classpathPrefix = classpathPrefix;
+  }
+
+  private final List<String> hadoopDependencyCoordinates;
+  private final URI distributedSuccessCache;
+  private final String jobPriority;
+  private final String segmentOutputPath;
+  private final String classpathPrefix;
+
+  @JsonProperty
+  public List<String> getHadoopDependencyCoordinates()
+  {
+    return hadoopDependencyCoordinates;
+  }
+
+  @JsonProperty
+  public URI getDistributedSuccessCache()
+  {
+    return distributedSuccessCache;
+  }
+
+  @JsonProperty
+  public String getJobPriority()
+  {
+    return jobPriority;
+  }
+
+  @JsonProperty
+  public String getSegmentOutputPath()
+  {
+    return segmentOutputPath;
+  }
+
+  @Override
+  @JsonProperty
+  public String getClasspathPrefix()
+  {
+    return classpathPrefix;
+  }
+
+  @Override
+  protected Iterable<Task> generateSubTasks(
+      final String groupId,
+      final Iterable<DataSegment> segments,
+      final IndexSpec indexSpec,
+      final boolean force,
+      final boolean validate
+  )
+  {
+    return Collections.<Task>singleton(
+        new ConverterSubTask(
+            ImmutableList.copyOf(segments),
+            this
+        )
+    );
+  }
+
+  @Override
+  @JsonIgnore
+  public DataSegment getSegment()
+  {
+    throw new UOE("Sub-less data segment not supported for hadoop converter task. Specify interval and datasource instead");
+  }
+
+  @Override
+  public String getType()
+  {
+    return TYPE;
+  }
+
+  public static class ConverterSubTask extends HadoopTask
+  {
+    private final List<DataSegment> segments;
+    private final HadoopConverterTask parent;
+
+    @JsonCreator
+    public ConverterSubTask(
+        @JsonProperty("segments") List<DataSegment> segments,
+        @JsonProperty("parent") HadoopConverterTask parent
+    )
+    {
+      super(
+          joinId(
+              Preconditions.checkNotNull(parent, "parent").getGroupId(),
+              "sub",
+              parent.getInterval().getStart(),
+              parent.getInterval().getEnd()
+          ),
+          parent.getDataSource(),
+          parent.getHadoopDependencyCoordinates()
+      );
+      this.segments = segments;
+      this.parent = parent;
+    }
+
+    @JsonProperty
+    public List<DataSegment> getSegments()
+    {
+      return segments;
+    }
+
+    @JsonProperty
+    public HadoopConverterTask getParent()
+    {
+      return parent;
+    }
+
+    @Override
+    public String getType()
+    {
+      return TYPE + "_sub";
+    }
+
+    @Override
+    public boolean isReady(TaskActionClient taskActionClient) throws Exception
+    {
+      return false;
```

Comment:
The problem is that HadoopTask cannot extend AbstracFixedIntervalTask because HadoopIndexerTask has multiple intervals definable in the GranularitySpec.